### PR TITLE
Modernize stack, add nap schedule + age-based recommendations

### PR DESCRIPTION
**Pull request by jfreal**
## Summary

Brings the feature branch up for review. Modernizes the build/stack, fixes a batch of scheduling bugs, and adds age-aware sleep recommendations with a nap schedule and responsive layout.

## What's in here
- **Stack modernization** — Vue 3.5 / Vite 6 / TS 5.9 / Tailwind v4 (via `@tailwindcss/vite`, no PostCSS config), Vitest 4. Netlify deploy fixed (Node 20).
- **Scheduling fixes** — corrected the flipped night-sleep sign and its follow-ons; `naps` getter now used in validation; `currentBracket` typed nullable with duplicate out-of-range messaging removed.
- **Recommendations** — age-bracketed guidance from multiple sources (Twins/Triplets/Quads, Huckleberry, AAP/Sleep Foundation), each cited with external links; reorganized comparison UI.
- **Nap schedule + layout** — evenly-split nap clock times between wake windows; responsive two-column layout.
- **NaN fix (latest)** — clearing the birthday field produced an Invalid Date, so the age getters returned `NaN` (`Math.max(0, NaN)` is `NaN`), cascading to "NaN weeks / NaN months" and "No guidance for NaN months". Both getters now return `0` for an empty/invalid birthday, with regression tests.

## Testing
- `npm test` — 49 passing (includes new empty-birthday cases).
- Verified in the browser: clearing the birthday now shows "0 weeks / 0 months", no `NaN` on the page, no console errors.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **Bug Fixes**
  * Age calculation methods now reliably handle edge cases, including empty birthday values, and return correct results instead of unexpected values.
  * Optimized internal performance of age computations.

* **Tests**
  * Added test coverage for age calculation edge cases to ensure accurate results under various conditions.

### CODE REVIEW

File: src/models/ScheduleSetting.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@ class ScheduleSetting {
 
     /** Adjusted age in weeks, accounting for gestational age (premature babies get a younger adjusted age). */
     public get weeksSinceBirth(): number {
+        if (isNaN(this.birthday.getTime())) return 0; // no/invalid birthday entered yet
         const msInWeek = 1000 * 60 * 60 * 24 * 7;
         const chronologicalWeeks = (new Date().getTime() - this.birthday.getTime()) / msInWeek;
         const gestationalAdjustment = 40 - this.weeks;
```
[the remark this answers is not in the document]
Comment (jfreal) [reply src/models/ScheduleSetting.ts]: Done in 398ee36 — cache the birthday ms once and use `Date.now()`; guard switched to `Number.isNaN`.

File: src/models/ScheduleSetting.ts [hunk 2]
```diff
@@ -44,6 +45,7 @@ class ScheduleSetting {
 
     /** Adjusted age in months, accounting for gestational age. */
     public get monthsSinceBirth(): number {
+        if (isNaN(this.birthday.getTime())) return 0; // no/invalid birthday entered yet
         const chronologicalMonths =
             (new Date().getFullYear() - this.birthday.getFullYear()) * 12
             + new Date().getMonth() - this.birthday.getMonth();
```
[the remark this answers is not in the document]
Comment (jfreal) [reply src/models/ScheduleSetting.ts]: Done in 398ee36 — snapshot a single `now` Date so the year/month reads can't cross a midnight boundary; guard switched to `Number.isNaN`.
